frontend/src/features/properties/map/acquisition/add/AddAcquisitionForm.tsx

```diff
@@ -97,6 +119,21 @@ export const AddAcquisitionForm = React.forwardRef<
             when={formikProps.dirty && formikProps.submitCount === 0}
             message="You have made changes on this form. Do you wish to leave without saving?"
           />
+
+          <AcquisitionFormModal
+            message="Selected Ministry region is different from that of one or more selected properties. Do you wish to continue?"
+            title="Different Ministry region"
+            display={showDiffMinistryRegionModal}
+            handleOk={() => {
+              setIgnoreRegionWarning(true);
```

Comment:
I'm not convinced this is safe. Essentially what you are doing here is
1) starting a new render because of a state change
2) starting another new render because of a state change
3) submitting the formik form.

I'm not convinced that there are any hard guarantees here that the above two renders will take place before the form submission which may result in a race condition. What I would recommend instead is that rather then using the modal to resubmit the form, store the form values either in a scoped arrow function or in state, and then submit those form values when the modal is confirmed.

Reply:
I over complicated it, we have to any way hide the modal using the state which will start a render. But I have removed the "ignoreRegionWarning" state change as it was unnecessary and directly used the onSubmit call. Form will submitted only once in this case.